### PR TITLE
add dtype information for input

### PR DESCRIPTION
**Pull request by guotuofeng**
add dtype for all input besides input dimenstion.

### CODE REVIEW

File: torch/csrc/autograd/profiler_legacy.cpp [hunk 1]
```diff
@@ -368,6 +368,24 @@ std::vector<std::vector<int64_t>> inputSizes(const at::RecordFunction& fn) {
   return sizes;
 }
 
+std::vector<std::string> inputTypes(const at::RecordFunction& fn) {
```
Comment (ilia-cher) [inline torch/csrc/autograd/profiler_legacy.cpp:371]: could you move this function to profiler_kineto.h/cpp ?

**Review by ilia-cher — APPROVED** [review]

**Comment by ilia-cher** [comment]
Thank you, looks good, modulo one minor comment. Also there might be some (trivial) merge conflict, so please rebase and we'll continue with landing.

File: torch/csrc/autograd/profiler_legacy.cpp (hunk 1 above)
Comment (guotuofeng) [reply torch/csrc/autograd/profiler_legacy.cpp:371]: will do

Comment (guotuofeng) [reply torch/csrc/autograd/profiler_legacy.cpp]: > Thank you, looks good, modulo one minor comment. Also there might be some (trivial) merge conflict, so please rebase and we'll continue with landing.

merged the latest code in master branch

File: torch/csrc/autograd/profiler_kineto.cpp [hunk 2]
```diff
@@ -174,6 +190,26 @@ struct TORCH_API KinetoThreadLocalState : public ProfilerThreadLocalState {
       std::make_unique<libkineto::CpuTraceBuffer>();
 };
 
+
+std::vector<std::string> inputTypes(const at::RecordFunction& fn) {
+  std::vector<std::string> types;
+  types.reserve(fn.inputs().size());
+  for (const c10::IValue& input : fn.inputs()) {
+    if (!input.isTensor()) {
+      types.emplace_back();
```
Comment (gdankel) [inline torch/csrc/autograd/profiler_kineto.cpp:199]: input.tagKind() ?

File: torch/csrc/autograd/profiler_kineto.cpp [hunk 3]
```diff
@@ -166,6 +171,17 @@ struct TORCH_API KinetoThreadLocalState : public ProfilerThreadLocalState {
         cpu_trace->activities[idx].callStack = stacksToStr(kineto_events_[idx].stack());
 #endif
       }
+#ifdef USE_KINETO_UPDATED
```
[the remark this answers is not in the document]
Comment (ilia-cher) [reply torch/csrc/autograd/profiler_kineto.cpp:174]: it's unrelated to this PR and caused by a temporary divergence between oss and internal kineto api (Jay's change), this macro will be removed very soon

File: torch/csrc/autograd/profiler_kineto.cpp [hunk 4]
```diff
@@ -262,6 +299,21 @@ std::string shapesToStr(const std::vector<std::vector<int64_t>>& shapes) {
   return oss.str();
 }
 
+std::string dtypesToStr(const std::vector<std::string>& types) {
+  std::ostringstream oss;
+  oss << "[";
```
Comment (gdankel) [inline torch/csrc/autograd/profiler_kineto.cpp:304]: I think you can use fmt::format() to simplify this?

Something like:

return fmt::format("["\{}\"]", types fmt::join(types, "\", \"));

File: torch/csrc/autograd/profiler_kineto.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (gdankel) [reply torch/csrc/autograd/profiler_kineto.cpp:174]: Never mind - I had missed this in Jay's earlier diff.

File: torch/csrc/autograd/profiler_kineto.cpp (hunk 4 above)
Comment (guotuofeng) [reply torch/csrc/autograd/profiler_kineto.cpp:304]: from the time being, the profiler_kineto doesn't include fmt dependency. Is it ok to introduce additional dependencies for all platforms?

Comment (ilia-cher) [reply torch/csrc/autograd/profiler_kineto.cpp:304]: yeah, try adding `#include <fmt/format.h>` i think it should work out of the box - pytorch already depends on fmt

File: torch/csrc/autograd/profiler_kineto.cpp (hunk 2 above)
Comment (guotuofeng) [reply torch/csrc/autograd/profiler_kineto.cpp:199]: From the time being, we only need care about scalar and tensor. What kind of scenarios that do we need care about list?

File: torch/csrc/autograd/profiler_kineto.cpp (hunk 4 above)
Comment (guotuofeng) [reply torch/csrc/autograd/profiler_kineto.cpp:304]: just find another way to simply the logic by using ostringstream without introduce the fmt dependency

Comment (guotuofeng) [reply torch/csrc/autograd/profiler_kineto.cpp]: it seems fmt::join fail in [pytorch_linux_xenial_py3_clang5_asan_test2]( https://circleci.com/api/v1.1/project/github/pytorch/pytorch/12606907/output/107/0?file=true&allocation-id=607e76b550c532340a9ce1c6-0-build%2F1E72253E)

File: torch/csrc/autograd/profiler_kineto.cpp [hunk 5]
```diff
@@ -166,6 +171,17 @@ struct TORCH_API KinetoThreadLocalState : public ProfilerThreadLocalState {
         cpu_trace->activities[idx].callStack = stacksToStr(kineto_events_[idx].stack());
 #endif
       }
+#ifdef USE_KINETO_UPDATED
+      if (kineto_events_[idx].hasTypes()) {
+        cpu_trace->activities[idx].addMetadata("Input type", dtypesToStr(kineto_events_[idx].dtypes());
```
Comment (ilia-cher) [inline torch/csrc/autograd/profiler_kineto.cpp:176]: missing ')' here and below, we will remove USE_KINETO_UPDATED usage soon, so one more rebase will be needed

**Pull request closed without merging**